### PR TITLE
Add common cache strategy

### PR DESCRIPTION
**Pull request by sbrunner**
See commits status

### CODE REVIEW

File: doc/developer/cache.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
+.. _developer_cache:
+
+Cache management
+================
+
+Here we describe the strategy we use to manage the cache.
+
+We have tree kinds of service:
+ * Without any cache (**no_cache**).
+ * Cache that's related of the version of the application (**public_cache**).
```
Comment (elemoine) [inline doc/developer/cache.rst]: of -> on

**Comment by sbrunner** [comment]
Ready to be reviewed !

**Comment by asaunier** [comment]
Typo in commit b458f74 title:
"Use tree type of cache" => "Use three types of cache"

**Comment by asaunier** [comment]
I have made a few small remarks inline. Else it looks good to me.

**Comment by elemoine** [comment]
@sbrunner, @ochriste and I have question about https://github.com/camptocamp/c2cgeoportal/commit/8f6fe38057397b24303bf3f095932158895cda58 and the `C2CPregenerator`. Why does the pregenerator add a `role` parameter to generated URLs? Is this just related to caching (by Varnish and/or the browser)? Or are there actual users of that `role` parameter?

**Comment by sbrunner** [comment]
It's only for the cache !

**Pull request merged**